{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b7ac2a7-553a-4982-a08c-b39215ee3669",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc\n",
    "import seaborn as sns\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import random\n",
    "import os\n",
    "from matplotlib.pyplot import rc_context\n",
    "sc.set_figure_params(dpi=100)\n",
    "\n",
    "import warnings\n",
    "warnings.simplefilter(\"ignore\", FutureWarning)\n",
    "warnings.simplefilter(\"ignore\", UserWarning)\n",
    "warnings.simplefilter(\"ignore\", RuntimeWarning)\n",
    "# Set working directory\n",
    "\n",
    "os.chdir(\"P:/Tolulope/Cellranger Results/YNO\")\n",
    "adata = sc.read_h5ad('integrated OLD.h5ad')\n",
    "adata"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d5b029dc-ff0e-4d09-a00a-db9ef024e6b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Subset for endothelial cells\n",
    "cell_subset = adata[adata.obs['cell type'] == \"Endothelial cells\"].copy()  # Make sure to use .copy() to avoid modifying original object\n",
    "from scipy.sparse import csr_matrix\n",
    "# Convert to sparse matrix\n",
    "cell_subset.X = csr_matrix(cell_subset.X)\n",
    "cell_subset.X\n",
    "\n",
    "#example WITH pseudo replicates\n",
    "pbs = []\n",
    "for sample in cell_subset.obs.Sample.unique():\n",
    "    samp_cell_subset = cell_subset[cell_subset.obs['Sample'] == sample]\n",
    "    \n",
    "    samp_cell_subset.X = samp_cell_subset.layers['counts'] #make sure to use raw data\n",
    "    \n",
    "    \n",
    "    \n",
    "    indices = list(samp_cell_subset.obs_names)\n",
    "    random.shuffle(indices)\n",
    "    indices = np.array_split(np.array(indices), 4) #change number here for number of replicates deisred\n",
    "    \n",
    "    for i, pseudo_rep in enumerate(indices):\n",
    "    \n",
    "        rep_adata = sc.AnnData(X = samp_cell_subset[indices[i]].X.sum(axis = 0),\n",
    "                               var = samp_cell_subset[indices[i]].var[[]])\n",
    "\n",
    "        rep_adata.obs_names = [sample + '_' + str(i)]\n",
    "        rep_adata.obs['condition'] = samp_cell_subset.obs['condition'].iloc[0]\n",
    "        rep_adata.obs['replicate'] = i\n",
    "\n",
    "        pbs.append(rep_adata)\n",
    "\n",
    "pb = sc.concat(pbs)\n",
    "pb.obs\n",
    "\n",
    "\n",
    "from pydeseq2.dds import DeseqDataSet\n",
    "from pydeseq2.ds import DeseqStats\n",
    "counts = pd.DataFrame(pb.X, columns = pb.var_names) #need to do this to pass var names\n",
    "\n",
    "# Create DeseqDataSet object with pseudo-replicates\n",
    "dds = DeseqDataSet(\n",
    "    counts=counts,  # Make sure 'counts' is correctly defined\n",
    "    metadata=pb.obs,\n",
    "    design_factors=['condition', 'replicate']  # Adjust to 'pseudo_replicate_group' if needed\n",
    ")\n",
    "sc.pp.filter_genes(dds, min_cells = 1)\n",
    "dds.deseq2()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4e4e5dec-3697-4f79-967b-38ee0d6a1253",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DGE Analysis of Young Endothelial Cells (YSham-GFP vs. YSham-noGFP) at Rest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fddf8a2-a646-41a4-9735-14fb9139adb3",
   "metadata": {},
   "outputs": [],
   "source": [
    "stat_res = DeseqStats(dds, contrast=('condition', 'YSham-GFP', 'YSham-noGFP'))\n",
    "stat_res.summary()\n",
    "de  = stat_res.results_df\n",
    "de.sort_values('stat', ascending = False)\n",
    "# Assuming 'res' is your DataFrame\n",
    "de['Symbol'] = de.index\n",
    "# Make 'Symbol' column uppercase\n",
    "de['Symbol'] = de['Symbol'].str.upper()\n",
    "de\n",
    "\n",
    "# Assuming 'de' is your DataFrame\n",
    "de_sorted = de.sort_values('stat', ascending=False)\n",
    "\n",
    "# Save the DataFrame with gene names as the index\n",
    "de_sorted.to_csv('YSham Endothelial cell1.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f87be4cf-c1cf-4ea4-b445-4f1899432d3b",
   "metadata": {},
   "outputs": [],
   "source": [
    "res = stat_res.results_df\n",
    "res\n",
    "# Assuming 'res' is your DataFrame\n",
    "res['Symbol'] = res.index\n",
    "\n",
    "# Print the DataFrame with gene symbols\n",
    "print(res)\n",
    "res\n",
    "res = res[res.baseMean >= 100]\n",
    "res\n",
    "sigs = res[(res.padj < 0.05) & (abs(res.log2FoldChange) > 0.5)]\n",
    "sigs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aee9256d-f2d3-4a9e-a3fa-18bed7286a45",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import seaborn as sns\n",
    "dds.layers['normed_counts']\n",
    "dds.layers['log1p'] = np.log1p(dds.layers['normed_counts'])\n",
    "dds.layers['log1p']\n",
    "sigs\n",
    "dds_sigs = dds[:, sigs.index]\n",
    "dds_sigs\n",
    "grapher = pd.DataFrame(dds_sigs.layers['log1p'].T,\n",
    "                       index=dds_sigs.var_names, columns=dds_sigs.obs_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88104864-3ef9-40fb-8954-02abe6dd0a1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Confirm and select columns of interest\n",
    "conditions_of_interest = ['YSham_GFP_0', 'YSham_GFP_1', 'YSham_GFP_2', 'YSham_GFP_3',\n",
    "                          'YSham_noGFP_0', 'YSham_noGFP_1', 'YSham_noGFP_2', 'YSham_noGFP_3']\n",
    "\n",
    "# Subset grapher DataFrame for significant genes and conditions of interest\n",
    "grapher_subset = grapher.loc[sigs.index.intersection(grapher.index), conditions_of_interest]\n",
    "\n",
    "# Create and display the clustermap\n",
    "plt.figure(figsize=(10, 8))  # Adjust figure size as needed\n",
    "clustermap = sns.clustermap(grapher_subset, z_score=0, cmap='RdYlBu_r')\n",
    "clustermap.ax_heatmap.set_xlabel('Samples')\n",
    "clustermap.ax_heatmap.set_ylabel('Genes')\n",
    "plt.title('Heatmap of Significant Genes between YSham_GFP vs YSham_noGFP')\n",
    "plt.savefig('clustermap_YSham.png')  # Save the clustermap as an image file\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e10ed9c9-d537-41b4-ab36-a5379e4a8a14",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "de2aaac1-86fc-49f1-b627-80d5b4afb982",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DGE Analysis of Aged Endothelial Cells (ASham-GFP vs. ASham-noGFP) at Rest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f542fbb-6941-4411-93f4-48d8ac5a5fcc",
   "metadata": {},
   "outputs": [],
   "source": [
    "stat_res = DeseqStats(dds, contrast=('condition', 'ASham-GFP', 'ASham-noGFP'))\n",
    "stat_res.summary()\n",
    "de  = stat_res.results_df\n",
    "de.sort_values('stat', ascending = False)\n",
    "# Assuming 'res' is your DataFrame\n",
    "de['Symbol'] = de.index\n",
    "# Make 'Symbol' column uppercase\n",
    "de['Symbol'] = de['Symbol'].str.upper()\n",
    "de\n",
    "\n",
    "# Assuming 'de' is your DataFrame\n",
    "de_sorted = de.sort_values('stat', ascending=False)\n",
    "\n",
    "# Save the DataFrame with gene names as the index\n",
    "de_sorted.to_csv('ASham Endothelial cell2.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "848ab52b-aab0-47a0-982c-ad3095901c80",
   "metadata": {},
   "outputs": [],
   "source": [
    "res = stat_res.results_df\n",
    "res\n",
    "# Assuming 'res' is your DataFrame\n",
    "res['Symbol'] = res.index\n",
    "\n",
    "# Print the DataFrame with gene symbols\n",
    "print(res)\n",
    "res\n",
    "res = res[res.baseMean >= 80]\n",
    "res\n",
    "sigs = res[(res.padj < 0.05) & (abs(res.log2FoldChange) > 0.5)]\n",
    "sigs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "31f22a9e-9347-4c23-ab2d-846bb9c97664",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import seaborn as sns\n",
    "dds.layers['normed_counts']\n",
    "dds.layers['log1p'] = np.log1p(dds.layers['normed_counts'])\n",
    "dds.layers['log1p']\n",
    "sigs\n",
    "dds_sigs = dds[:, sigs.index]\n",
    "dds_sigs\n",
    "grapher = pd.DataFrame(dds_sigs.layers['log1p'].T,\n",
    "                       index=dds_sigs.var_names, columns=dds_sigs.obs_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d8b1944-3c79-4951-8fe6-b186aced1799",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Confirm and select columns of interest\n",
    "conditions_of_interest = ['ASham_GFP_0', 'ASham_GFP_1', 'ASham_GFP_2', 'ASham_GFP_3',\n",
    "                          'ASham_noGFP_0', 'ASham_noGFP_1', 'ASham_noGFP_2', 'ASham_noGFP_3']\n",
    "\n",
    "# Subset grapher DataFrame for significant genes and conditions of interest\n",
    "grapher_subset = grapher.loc[sigs.index.intersection(grapher.index), conditions_of_interest]\n",
    "\n",
    "# Create and display the clustermap\n",
    "plt.figure(figsize=(10, 8))  # Adjust figure size as needed\n",
    "clustermap = sns.clustermap(grapher_subset, z_score=0, cmap='RdYlBu_r')\n",
    "clustermap.ax_heatmap.set_xlabel('Samples')\n",
    "clustermap.ax_heatmap.set_ylabel('Genes')\n",
    "plt.title('Heatmap of Significant Genes between ASham_GFP vs ASham_noGFP')\n",
    "plt.savefig('clustermap_ASham.png')  # Save the clustermap as an image file\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "682a97a7-4a49-428f-a298-07699deaf1bf",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "34796600-f420-41e3-ac2e-5eed7b1250bc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#DGE Analysis of Young and Aged GFP-Positive Endothelial Cells at Rest"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cfc5b794-8f74-448f-a486-c64d0489fce4",
   "metadata": {},
   "outputs": [],
   "source": [
    "stat_res = DeseqStats(dds, contrast=('condition', 'YSham-GFP', 'ASham-GFP'))\n",
    "stat_res.summary()\n",
    "de  = stat_res.results_df\n",
    "de.sort_values('stat', ascending = False)\n",
    "# Assuming 'res' is your DataFrame\n",
    "de['Symbol'] = de.index\n",
    "# Make 'Symbol' column uppercase\n",
    "de['Symbol'] = de['Symbol'].str.upper()\n",
    "de\n",
    "\n",
    "# Assuming 'de' is your DataFrame\n",
    "de_sorted = de.sort_values('stat', ascending=False)\n",
    "\n",
    "# Save the DataFrame with gene names as the index\n",
    "de_sorted.to_csv('Y&ASham Endothelial cell3.csv')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c0e15a8-52f0-49ee-b315-84d42960e668",
   "metadata": {},
   "outputs": [],
   "source": [
    "res = stat_res.results_df\n",
    "res\n",
    "# Assuming 'res' is your DataFrame\n",
    "res['Symbol'] = res.index\n",
    "\n",
    "# Print the DataFrame with gene symbols\n",
    "print(res)\n",
    "res\n",
    "res = res[res.baseMean >= 80]\n",
    "res\n",
    "sigs = res[(res.padj < 0.05) & (abs(res.log2FoldChange) > 0.5)]\n",
    "sigs\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8b7bba98-ff8a-474d-8936-6a4feb9c7055",
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import seaborn as sns\n",
    "dds.layers['normed_counts']\n",
    "dds.layers['log1p'] = np.log1p(dds.layers['normed_counts'])\n",
    "dds.layers['log1p']\n",
    "sigs\n",
    "dds_sigs = dds[:, sigs.index]\n",
    "dds_sigs\n",
    "grapher = pd.DataFrame(dds_sigs.layers['log1p'].T,\n",
    "                       index=dds_sigs.var_names, columns=dds_sigs.obs_names)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ea85801-6674-44da-b8e7-bbad38d76842",
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "\n",
    "# Confirm and select columns of interest\n",
    "conditions_of_interest = ['YSham_GFP_0', 'YSham_GFP_1', 'YSham_GFP_2', 'YSham_GFP_3',\n",
    "                          'ASham_GFP_0', 'ASham_GFP_1', 'ASham_GFP_2', 'ASham_GFP_3']\n",
    "\n",
    "# Subset grapher DataFrame for significant genes and conditions of interest\n",
    "grapher_subset = grapher.loc[sigs.index.intersection(grapher.index), conditions_of_interest]\n",
    "\n",
    "# Create and display the clustermap\n",
    "plt.figure(figsize=(10, 8))  # Adjust figure size as needed\n",
    "clustermap = sns.clustermap(grapher_subset, z_score=0, cmap='RdYlBu_r')\n",
    "clustermap.ax_heatmap.set_xlabel('Samples')\n",
    "clustermap.ax_heatmap.set_ylabel('Genes')\n",
    "plt.title('Heatmap of Significant Genes between YSham_GFP vs ASham_GFP')\n",
    "plt.savefig('clustermap_ASham.png')  # Save the clustermap as an image file\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "283a52e7-fee4-465f-a580-3fe00f6a3ffb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ca35dcf-55b1-4a0e-bb5e-3418db563792",
   "metadata": {},
   "outputs": [],
   "source": [
    "# The above code was reused for differential gene expression (DGE) analysis  \n",
    "# of other cell types, including:  \n",
    "# - Anti-inflammatory macrophages at rest  \n",
    "# - Lipid-laden macrophages after mechanical overload  \n",
    "# - Pro-inflammatory macrophages II during MOV  \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "33f72abb-717d-4f59-8617-3fb2dcaf4c16",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ffb5f24a-ead4-41d8-bdef-47ee3270c0a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Venn diagram: Overlapping and unique differentially expressed genes (DEGs)  \n",
    "# between young and aged endothelial cells (GFP+ vs. GFP−) AT REST.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "98789935-81bd-42c8-ad6b-920650b3fbc9",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib_venn import venn2\n",
    "import os\n",
    "\n",
    "# Set working directory\n",
    "os.chdir(\"P:/Tolulope/Cellranger Results/YNO/excel\")\n",
    "\n",
    "# Load DEG data from CSV files\n",
    "df_ysham = pd.read_csv('YSham Endothelial cells.csv')\n",
    "df_asham = pd.read_csv('ASham Endothelial cells.csv')\n",
    "\n",
    "# Ensure 'Symbol' column is in uppercase\n",
    "df_ysham['Symbol'] = df_ysham['Symbol'].str.upper()\n",
    "df_asham['Symbol'] = df_asham['Symbol'].str.upper()\n",
    "\n",
    "# Define the conditions for upregulated and downregulated genes in each dataset\n",
    "up_genes_ysham = set(df_ysham[(df_ysham['padj'] < 0.05) & (df_ysham['log2FoldChange'] > 0.5)]['Symbol'])\n",
    "down_genes_ysham = set(df_ysham[(df_ysham['padj'] < 0.05) & (df_ysham['log2FoldChange'] < -0.5)]['Symbol'])\n",
    "\n",
    "up_genes_asham = set(df_asham[(df_asham['padj'] < 0.05) & (df_asham['log2FoldChange'] > 0.5)]['Symbol'])\n",
    "down_genes_asham = set(df_asham[(df_asham['padj'] < 0.05) & (df_asham['log2FoldChange'] < -0.5)]['Symbol'])\n",
    "\n",
    "# Create Venn diagrams for upregulated and downregulated genes\n",
    "plt.figure(figsize=(12, 6))\n",
    "\n",
    "# Upregulated Genes\n",
    "plt.subplot(1, 2, 1)\n",
    "venn2([up_genes_ysham, up_genes_asham], ('YSham Upregulated', 'ASham Upregulated'))\n",
    "plt.title('Upregulated Genes Comparison')\n",
    "\n",
    "# Downregulated Genes\n",
    "plt.subplot(1, 2, 2)\n",
    "venn2([down_genes_ysham, down_genes_asham], ('YSham Downregulated', 'ASham Downregulated'))\n",
    "plt.title('Downregulated Genes Comparison')\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c591bf0-3605-4a2b-9b33-5ab6dd22b0f6",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c71f2912-a262-49a9-ab06-50b702597af9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Venn diagram: Overlapping and unique differentially expressed genes (DEGs)  \n",
    "# between young and aged Anti-inflammatory Macrophages (GFP+ vs. GFP−) AT REST.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8386d2b2-4c80-4300-9590-1dbeadb7c8d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib_venn import venn2\n",
    "import os\n",
    "\n",
    "# Set working directory\n",
    "os.chdir(\"P:/Tolulope/Cellranger Results/YNO/excel\")\n",
    "\n",
    "# Load DEG data from CSV files\n",
    "df_ysham = pd.read_csv('YSham Anti-inflammatory Macrophages.csv')\n",
    "df_asham = pd.read_csv('ASham Anti-inflammatory Macrophages.csv')\n",
    "\n",
    "# Ensure 'Symbol' column is in uppercase\n",
    "df_ysham['Symbol'] = df_ysham['Symbol'].str.upper()\n",
    "df_asham['Symbol'] = df_asham['Symbol'].str.upper()\n",
    "\n",
    "# Define the conditions for upregulated and downregulated genes in each dataset\n",
    "up_genes_ysham = set(df_ysham[(df_ysham['padj'] < 0.05) & (df_ysham['log2FoldChange'] > 0.5)]['Symbol'])\n",
    "down_genes_ysham = set(df_ysham[(df_ysham['padj'] < 0.05) & (df_ysham['log2FoldChange'] < -0.5)]['Symbol'])\n",
    "\n",
    "up_genes_asham = set(df_asham[(df_asham['padj'] < 0.05) & (df_asham['log2FoldChange'] > 0.5)]['Symbol'])\n",
    "down_genes_asham = set(df_asham[(df_asham['padj'] < 0.05) & (df_asham['log2FoldChange'] < -0.5)]['Symbol'])\n",
    "\n",
    "# Create Venn diagrams for upregulated and downregulated genes\n",
    "plt.figure(figsize=(12, 6))\n",
    "\n",
    "# Upregulated Genes\n",
    "plt.subplot(1, 2, 1)\n",
    "venn2([up_genes_ysham, up_genes_asham], ('YSham Upregulated', 'ASham Upregulated'))\n",
    "plt.title('Upregulated Genes Comparison')\n",
    "\n",
    "# Downregulated Genes\n",
    "plt.subplot(1, 2, 2)\n",
    "venn2([down_genes_ysham, down_genes_asham], ('YSham Downregulated', 'ASham Downregulated'))\n",
    "plt.title('Downregulated Genes Comparison')\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e6245a68-10e6-4540-8f13-b2d8ca83ac7d",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8380bb4f-e2e1-4bc2-9b24-7fb70bb91005",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Venn diagram: Overlapping and unique differentially expressed genes (DEGs)  \n",
    "# between young and aged Lipid-Laden Macrophages (GFP+ vs. GFP−) durin MOV.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1191270-601c-4aef-b6f8-c1c6d763d4b7",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "from matplotlib_venn import venn2\n",
    "import os\n",
    "\n",
    "# Set working directory\n",
    "os.chdir(\"P:/Tolulope/Cellranger Results/YNO/excel\")\n",
    "\n",
    "# Load DEG data from CSV files\n",
    "df_ysham = pd.read_csv('YOV Lipid-Laden Macrophages.csv')\n",
    "df_asham = pd.read_csv('AOV Lipid-Laden Macrophages.csv')\n",
    "\n",
    "# Ensure 'Symbol' column is in uppercase\n",
    "df_ysham['Symbol'] = df_ysham['Symbol'].str.upper()\n",
    "df_asham['Symbol'] = df_asham['Symbol'].str.upper()\n",
    "\n",
    "# Define the conditions for upregulated and downregulated genes in each dataset\n",
    "up_genes_ysham = set(df_ysham[(df_ysham['padj'] < 0.05) & (df_ysham['log2FoldChange'] > 0.5)]['Symbol'])\n",
    "down_genes_ysham = set(df_ysham[(df_ysham['padj'] < 0.05) & (df_ysham['log2FoldChange'] < -0.5)]['Symbol'])\n",
    "\n",
    "up_genes_asham = set(df_asham[(df_asham['padj'] < 0.05) & (df_asham['log2FoldChange'] > 0.5)]['Symbol'])\n",
    "down_genes_asham = set(df_asham[(df_asham['padj'] < 0.05) & (df_asham['log2FoldChange'] < -0.5)]['Symbol'])\n",
    "\n",
    "# Create Venn diagrams for upregulated and downregulated genes\n",
    "plt.figure(figsize=(12, 6))\n",
    "\n",
    "# Upregulated Genes\n",
    "plt.subplot(1, 2, 1)\n",
    "venn2([up_genes_ysham, up_genes_asham], ('YSham Upregulated', 'ASham Upregulated'))\n",
    "plt.title('Upregulated Genes Comparison')\n",
    "\n",
    "# Downregulated Genes\n",
    "plt.subplot(1, 2, 2)\n",
    "venn2([down_genes_ysham, down_genes_asham], ('YSham Downregulated', 'ASham Downregulated'))\n",
    "plt.title('Downregulated Genes Comparison')\n",
    "\n",
    "plt.tight_layout()\n",
    "plt.show()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.21"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
